.github/workflows/reusable-run-integration-tests.yml

```diff
@@ -0,0 +1,129 @@
+        default: ${{ github.ref }}
+      go_version:
+        description: Golang version to use for running tests
+        required: false
+        type: string
+        default: "1.17"
+      go_test_params:
+        description: Parameters to pass to go tests
+        required: false
+        type: string
+        default: "-v -race -tags=integration ./..."
+      enable_kind:
+        description: Enable a kind cluster.
+        required: false
+        type: boolean
+        default: false
+      auth0_client_id:
+        description: client used for OIDC authentication.
+        required: false
+        type: string
+    secrets:
+      fluentbit_config_validator_api_key:
+        description: Fluentbit config validator API key
+        required: false
+      fluentd_config_validator_api_key:
+        description: Fluentd config validator API Key
+        required: false
+      smtp_username:
+        description: SMTP username to test email notifications
+        required: false
+      smtp_password:
+        description: SMTP password to test email notifications
+        required: false
+      image_registry_username:
+        description: Password (if github use a PAT token) to use for authentication on private registry
+        required: false
+      image_registry_password:
+        description: Password (if github use a PAT token) to use for authentication on private registry
+        required: false
+      auth0_client_secret:
+        description: Auth0 client secret
+        required: false
+jobs:
+  call-run-integration-test-images-exist:
+    name: Test image exists
+    runs-on: ubuntu-latest
+    env:
+      IMAGE: ${{ inputs.cloud_image_registry }}/${{ inputs.cloud_image_name }}:${{ inputs.cloud_image_tag }}
+    steps:
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ inputs.cloud_image_registry }}
+          username: ${{ secrets.image_registry_username }}
+          password: ${{ secrets.image_registry_password }}
+      - run: docker pull $IMAGE
+
+  call-run-integration-tests:
+    name: run integration tests
+    needs: call-run-integration-test-images-exist
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: ${{ inputs.target_repository }}
+          ref: ${{ inputs.target_repository_ref }}
+          token: ${{ secrets.image_registry_password }}
+          fetch-depth: 0
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ inputs.cloud_image_registry }}
+          username: ${{ secrets.image_registry_username }}
+          password: ${{ secrets.image_registry_password }}
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ inputs.go_version }}
+
+      - name: Create k8s Kind Cluster
+        if: ${{ inputs.enable_kind == true }}
+        uses: helm/kind-action@v1.2.0
```

Comment:
Do we want configurable options here for KIND (E.g. version of K8S, etc.)

Reply:
yeap, added kind_node_version string variable default to latest one.